src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -91,21 +79,105 @@ const DiffExprCompute = (props) => {
+
+    if (!basis || !cellSet || !compareWith || !cellIdToSampleMap.length > 0) { return false; }
+
+    const basisCellSetKey = getCellSetName(basis);
+
+    // Group 1 is from cellSet
+    // Group 2 is from compareWith
+    const group1CellSetKey = getCellSetName(cellSet);
+    const group2CellSetKey = getCellSetName(compareWith);
+
+    let basisCellSet = [];
+    if (basisCellSetKey === 'all') {
+      const allCellIds = sampleKeys.reduce((acc, key) => {
+        const cellIds = properties[key].cellIds;
+        return acc.concat(Array.from(cellIds));
+      }, []);
+      basisCellSet = new Set(allCellIds);
+    } else {
+      basisCellSet = properties[basisCellSetKey].cellIds;
+    }
+
+    const group1CellIds = Array.from(properties[group1CellSetKey].cellIds);
+
+    let group2CellIds = [];
+    if (['rest', 'background'].includes(group2CellSetKey)) {
+      const parentGroup = getRootKey(cellSet);
+
+      const otherGroupKeys = hierarchy.find(obj => obj.key === parentGroup)
+        .children.filter(child => child.key !== group1CellSetKey);
+
+      group2CellIds = otherGroupKeys.reduce((acc, child) => {
+        return acc.concat(Array.from(properties[child.key].cellIds));
+      }, []);
+    } else {
+      group2CellIds = Array.from(properties[group2CellSetKey].cellIds);
     }
 
-    if (!comparisonGroup[selectedComparison]?.compareWith) {
+    // Intersect the basis cell set with each group cell set
+    const filteredGroup1CellIds = group1CellIds.filter(cellId => basisCellSet.has(cellId));
+    const filteredGroup2CellIds = group2CellIds.filter(cellId => basisCellSet.has(cellId));
+
+    // Prepare an array of length sampleIds to hold tally of cells for each sapmple in each group
+    const cellsInGroup1Samples = new Array(numSamples).fill(0);
+    const cellsInGroup2Samples = new Array(numSamples).fill(0);
+
+    // Count the number of cells in each sample
+    filteredGroup1CellIds
+      .forEach(cellId => {
+        const sampleIdx = cellIdToSampleMap[cellId];
+        cellsInGroup1Samples[sampleIdx] += 1;
+      });
+
+    filteredGroup2CellIds
+      .forEach(cellId => {
+        const sampleIdx = cellIdToSampleMap[cellId];
+        cellsInGroup2Samples[sampleIdx] += 1;
+      });
+
+    console.log("cellsInGroup1Samples", cellsInGroup1Samples)
+    console.log("cellsInGroup2Samples", cellsInGroup2Samples)
+
+    // The samples in the two groups will always be exclusive, i.e. presence of a cell in a sample
+    // in group1 means there are no cells in group2 in that sample. Therefore, finding the first
+    // sample with the minimum number of cells in each will be enough to determine if there are at least 2
+    // different samples to compare with.
+    const MIN_NUM_CELLS = 10;
+    const group1Passes = cellsInGroup1Samples.find(numCells => numCells > MIN_NUM_CELLS);
+    const group2Passes = cellsInGroup2Samples.find(numCells => numCells > MIN_NUM_CELLS);
```

Comment:
I would change the names of `group1Passes` and `group2Passes` to be more clear on what "passes" means. For example in this case, it can be `group1HasMoreThanMinNumCells`

Reply:
changed